hi-ml-histopathology/src/histopathology/utils/wsi_utils.py

```diff
@@ -0,0 +1,20 @@
+import torch
+import numpy as np
+
+from typing import List
+from histopathology.utils.naming import SlideKey
+from torch.utils.data.dataloader import default_collate
+
+
+def list_data_collate(batch: List[dict]) -> List[dict]:
+    """
+        Combine instances from a list of dicts into a single dict, by stacking them along first dim
+        [{'image' : 3xHxW}, {'image' : 3xHxW}, {'image' : 3xHxW}...] - > {'image' : Nx3xHxW}
+        followed by the default collate which will form a batch BxNx3xHxW
+    """
+
+    for i, item in enumerate(batch):
+        data = item[0]
+        data[SlideKey.IMAGE] = torch.tensor(np.array([ix[SlideKey.IMAGE] for ix in item]))
```

Comment:
I think I had left this comment also in the previous draft PR, maybe among those you haven't seen: if this needs to be a generic function you shouldn't hardcode SlideKey.IMAGE here, you can pass the key as parameter of the function

Reply:
Yes, I left a comment on the previous PR as well, this function is used for collate_fn of torch Dataloader. If we add extra parameters, we should use partial(SlideKey.IMAGE) to specify the key, would that be ok?

Reply:
sorry I hadn't seen your reply there! It's OK we can leave it as custom function instead of generic one. Maye change the docstring to make that clear `Combine images from a list of dicts into a single dict`?